### PR TITLE
Kirill orlov hm 16

### PR DESCRIPTION
**Pull request by kirillorlov1205**

### CODE REVIEW

File: hw16/index.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (VladSelikh) [inline hw16/index.ts:29]: Please delete all these unnecessary line breaks.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    "url": "git+https://github.com/kirillorlov1205/JSAutomrationLessons.git"
+  },
+  "author": "",
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/kirillorlov1205/JSAutomrationLessons/issues"
+  },
+  "homepage": "https://github.com/kirillorlov1205/JSAutomrationLessons#readme",
+  "devDependencies": {
+    "ts-node": "^10.9.1"
```
Comment (VladSelikh) [inline package.json:20]: I think you are missing `typescript` package.

File: hw16/hw.md [hunk 3]
```diff
@@ -0,0 +1,2 @@
+- Реализовать функцию calculator, которая принимает принимает 2 параметра: 2 цифры и операция (сложение/вычитание/деление/умножение); и возвращает результат выполнения данной операции.
+- Написать дополнительную функцию, которая принимает те же аргументы + 4-й аргумент, который соответствует ожидаемому результату выполнения функции с указанными аргументами. В результате своего выполнения данная функция выводит в консоль сообщение о том, что "тест" либо прошёл, либо свалился.- В отдельном файле написать минимум 7 "положительных" тестов и минимум 7 "отрицательных" тестов. Тесты должны быть максимально осознанными, т.е. с использованием пройденных техник тест-дизайна.BONUS: усовершенствовать функцию calculator таким образом, чтобы помимо перечисленных выше операций пользователь мог совершать ещё одну (на ваше усмотрение) и написать на эту операцию соответствующие тесты.
```
Comment (VladSelikh) [inline hw16/hw.md:2]: You don't have any `.gitignore` file. Please add it.

File: hw16/src/calculationChecker.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import { calculator } from "./calculator"
+
+export function calculationChecker(num1: number, num2: number, operation: "+" | "-" | "/" | "*" | "**", result: number) {
+    (calculator(num1, num2, operation) === result) ?
+        console.log(`Result = ${result} - Test Passed`) :
+        console.log(`Result = ${calculator(num1, num2, operation)}, Expected result = ${result} - Test Failed`);
```
Comment (VladSelikh) [inline hw16/src/calculationChecker.ts:6]: I believe it should be vice versa:
```
`Result = ${result}, Expected result = ${calculator(num1, num2, operation)} - Test Failed`
```
Expected result is the value returned by a `calculator` function.

File: hw16/src/calculationChecker.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import { calculator } from "./calculator"
+
+export function calculationChecker(num1: number, num2: number, operation: "+" | "-" | "/" | "*" | "**", result: number) {
+    (calculator(num1, num2, operation) === result) ?
```
Comment (VladSelikh) [inline hw16/src/calculationChecker.ts:4]: Please create a `const` for this `calculator(num1, num2, operation)` function call, as you use this statement twice inside this function.

File: hw16/src/calculationChecker.ts (hunk 4 above)
Comment (kirillorlov1205) [reply hw16/src/calculationChecker.ts:6]: Expected result is the value we provide as a parameter to checker, I believe my implementation is correct.

File: package.json [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    "allure-mocha": "^2.0.0-beta.18",
+    "chai": "^4.3.6",
+    "jest": "^28.1.3",
+    "mocha": "^10.0.0",
+    "nyc": "^15.1.0",
+    "ts-jest": "^28.0.7",
+    "ts-node": "^10.9.1",
+    "typescript": "^4.7.4"
+  },
+  "description": ""
```
Comment (VladSelikh) [inline package.json:38]: You have some packages and scripts that are not used by now: 
`@types/jest`, `mocha`, etc.
`hw17-jest`, `allure-report`, etc.
Please delete them.

File: hw16/src/calculationChecker.ts [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import { calculator } from "./calculator"
+
+
+export function calculationChecker(num1: number, num2: number, operation: "+" | "-" | "/" | "*" | "**", result: number) {
+const test_calculator = calculator(num1, num2, operation);
```
Comment (VladSelikh) [inline hw16/src/calculationChecker.ts:5]: As we are writing on TypeScript, we are using camelNotation while namong variables.
Please give more descriptive name: `fucntionResult` for example.

**Review by VladSelikh — APPROVED** [review]

**Pull request merged**